codec.go

```diff
@@ -227,3 +230,19 @@ func (u *UUID) UnmarshalBinary(data []byte) error {
 
 	return nil
 }
+
+// WithCustomPRNG provides a deterministic random number generator for testing.
+//
+// Allows users to specify a PRNG with a fixed seed, enabling
+// reproducible UUID generation. Useful for unit testing and debugging.
+//
+// Arguments:
+// - seed: The seed value for the PRNG.
+//
+// Returns:
+// - GenOption: A function to configure the generator.
+func WithCustomPRNG(seed int64) GenOption {
```

Comment:
Assuming we move forward I think this function should be called `WithExplicitRandSeed()`, or something similar, to better reflect what it's actually doing.